### PR TITLE
Service tests for cancellation

### PR DESCRIPTION
**Pull request by fleetwoods**
Service tests now do front end tests
Limit docker-compose container sizes

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -25,6 +25,7 @@ services:
     environment:
       ZOOKEEPER_CLIENT_PORT: 32181
       ZOOKEEPER_TICK_TIME: 2000
+      KAFKA_HEAP_OPTS: "-Xmx256M -Xms128M"
```
Comment (cb372) [inline docker-compose.yml]: It's not doing any harm, but as far as I can tell this env var will have no effect for ZooKeeper

File: test/ServiceTestIT.scala [hunk 2]
```diff
@@ -98,12 +116,9 @@ class ServiceTestIT
     )
 
     sendEventAndTest(triggeredProducer, triggeredTopic, triggered) {
-      val query = table.limit(1).query('traceToken -> triggered.metadata.traceToken)
-      val result = Scanamo.exec(dynamoClient)(query).head
-      val dbResult = result.right.value
-
-      val events = result.right.value.events
-      val comm = Comm.fromDbModel(dbResult)
+      val res = getCommAndEventsFromDynamo(triggered.metadata.traceToken)
```
Comment (cb372) [inline test/ServiceTestIT.scala]: could rewrite in one line as 

```scala
val (comm, events) = getCommAndEventsFromDynamo(triggered.metadata.traceToken)
```

File: test/ServiceTestIT.scala [hunk 3]
```diff
@@ -214,6 +220,153 @@ class ServiceTestIT
...
     }
+
+    validateDisplayedResult(
+      searchCommsOnGui(Some(composedMetadata.commManifest.name), None, None, composedMetadata.traceToken))
+    validateDisplayedResult(
+      searchCommsOnGui(None, Some(composedMetadata.customerId), None, composedMetadata.traceToken))
+    validateDisplayedResult(searchCommsOnGui(None, None, Some("Delivered"), composedMetadata.traceToken))
+
+    def validateDisplayedResult(result: Option[CommsSearchResult]) = {
+      withScreenshot {
```
Comment (cb372) [inline test/ServiceTestIT.scala]: Where do the screenshots get saved? Let's save them as artifacts in CircleCI: https://circleci.com/docs/1.0/configuration/#artifacts

File: docker-compose.yml (hunk 1 above)
Comment (fleetwoods) [inline docker-compose.yml]: As discussed in IRL they work for some reason

**Comment by ovo-comms-circleci** [comment]
This PR was successfully deployed to production by [CircleCI build 144](https://circleci.com/gh/ovotech/comms-audit-log/144) at 2017-03-16 17:57 GMT.

**Pull request merged**